internal/pkg/doc/input.go

```diff
@@ -53,3 +53,32 @@ func (a inputsSortedByName) Less(i, j int) bool {
 func SortInputsByName(inputs []Input) {
 	sort.Sort(inputsSortedByName(inputs))
 }
+
+type inputsSortedByRequired []Input
+
+func (a inputsSortedByRequired) Len() int {
+	return len(a)
+}
+
+func (a inputsSortedByRequired) Swap(i, j int) {
+	a[i], a[j] = a[j], a[i]
+}
+
+func (a inputsSortedByRequired) Less(i, j int) bool {
+	switch {
+	// i required, j not: i gets priority
+	case !a[i].HasDefault() && a[j].HasDefault():
+		return true
+		// j required, i not: i does not get priority
```

Comment:
The comment is not proper indented. Are you using `go fmt`?

Reply:
Hmmm, gofmt didn't pick that one up, I guess because it's inline with the item above!